lib/fizzy/parser.hpp

```diff
@@ -24,40 +24,6 @@ parser_result<Code> parse_expr(const uint8_t* input);
 template <typename T>
 parser_result<T> parse(const uint8_t* pos);
 
-template <typename T>
-struct parser
-{
-};
-
-template <>
-inline parser_result<uint8_t> parse(const uint8_t* pos)
-{
-    const auto result = *pos;
-    ++pos;
-    return {result, pos};
-}
-
-template <>
-inline parser_result<uint32_t> parse(const uint8_t* pos)
-{
-    return leb128u_decode<uint32_t>(pos);
-}
-
-template <>
-inline parser_result<ValType> parse(const uint8_t* pos)
-{
-    const auto b = *pos++;
-    switch (b)
-    {
-    case 0x7F:
-        return {ValType::i32, pos};
-    case 0x7E:
-        return {ValType::i64, pos};
-    default:
-        throw parser_error{"invalid valtype " + std::to_string(b)};
-    }
-}
-
 inline parser_result<Limits> parse_limits(const uint8_t* pos)
```

Comment:
How about this?

Reply:
This is used in unit tests directly.

Reply:
Yes, but will the test be changed (in a new PR)?

Reply:
Can be done, but adding bounds checking to parser seems more important at the moment.